packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -188,12 +245,32 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
       throw new Error('At least one of \'port\' or \'protocol\' is required');
     }
 
+    if (props.mutualAuthentication?.mutualAuthenticationMode === MutualAuthenticationMode.VERIFY && !props.mutualAuthentication.trustStore) {
+      throw new Error(`You must set 'trustStore' when 'mode' is '${MutualAuthenticationMode.VERIFY}'`);
+    }
+
+    if ((props.mutualAuthentication?.mutualAuthenticationMode === MutualAuthenticationMode.OFF ||
+      props.mutualAuthentication?.mutualAuthenticationMode === MutualAuthenticationMode.PASS_THROUGH)) {
+      if (props.mutualAuthentication.trustStore) {
+        throw new Error(`You cannot set 'trustStore' when 'mode' is '${MutualAuthenticationMode.OFF}' or '${MutualAuthenticationMode.PASS_THROUGH}'`);
+      }
+
+      if (props.mutualAuthentication.ignoreClientCertificateExpiry !== undefined) {
+        throw new Error(`You cannot set 'ignoreClientCertificateExpiry' when 'mode' is '${MutualAuthenticationMode.OFF}' or '${MutualAuthenticationMode.PASS_THROUGH}'`);
+      }
+    }
```

Comment:
I feel like this is somewhat tricky to follow. Maybe having some variable for the current mode would help? Something like:
```ts
currentMode: MutualAuthenticationMode = props.mutualAuthentication?.mutualAuthenticationMode
```

Reply:
Thanks. Updated.